WorkoutTest/ExerciseController.swift

```diff
@@ -167,7 +167,16 @@ class ExerciseController: UIViewController {
   }
   
   private func removeTimerView() {
-    timerView.removeFromSuperview()
+    
+    UIView.animate(withDuration: 0.5, delay: 0.1, animations: {
+      self.nextButton.frame.origin.y -= 100
+      self.backButton.frame.origin.y -= 100
+      self.timerView.alpha = 0
+    }, completion: { (finished) in
+      if finished {
+        self.timerView.removeFromSuperview()
```

Comment:
while it's removing the view from the view hierarchy - the instance of this view is still alive in memory;  You should make timerView optional and set it to nil after removing from hierarchy